### DISCUSSION TITLE
Build rust project with tree-sitter-tlaplus to wasm target

### DISCUSSION THREAD

**Discussion by FedericoPonzi** [tlaplus-community/tree-sitter-tlaplus#122] [Q&A]
I'd like to write a program in rust using tree-sitter-tlaplus and compile it to wasm. When I try to compile it to wasm it fails with:
```
[...]
  CFLAGS = None
  cargo:warning=In file included from /home/fponzi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tree-sitter-0.22.6/src/lib.c:3:
  cargo:warning=In file included from /home/fponzi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tree-sitter-0.22.6/src/./alloc.c:1:
  cargo:warning=/home/fponzi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tree-sitter-0.22.6/src/alloc.h:9:10: fatal error: 'stdio.h' file not found
  cargo:warning=#include <stdio.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=1 error generated.
[...]
```
Is it possible to compile it to wasm?

**Comment by ahelwer** [comment]
This is something I wanted to do with [tlauc](https://github.com/tlaplus-community/tlauc) 1.5 years ago so it could have a web demo. Unfortunately I never figured it out. Here are some questions I asked about it at the time, with some possible starts to a solution:

- https://github.com/tree-sitter/tree-sitter/discussions/2010
- https://stackoverflow.com/questions/75025716/can-wasm-pack-compile-a-rust-project-including-c-code-that-uses-stdlib

My impression is that Rust-WASM-C++ development has been ongoing so maybe the toolchain has improved by now. If you figure it out please let me know!
